lib/src/realm_class.dart

```diff
@@ -386,6 +310,19 @@ class Realm {
     if (other is! Realm) return false;
     return realmCore.realmEquals(this, other);
   }
+
+  /// The logger to use for logging or null. 
+  /// 
+  /// To enable logging assign a logger instance and recreate any [FlexibleSyncConfiguration] configuraitons and reopen the synced [Realm]s.
+  static Logger? logger;
```

Comment:
I think we should default to log stuff. Also, I don't like this being nullable. We could choose use a default level of `RealmLogLevel.off` in release mode.

Reply:
see my other comment

Reply:
Here I am not sure I understand what don't you like about it. 

What do you mean "in release mode". Can you elaborate?

Reply:
```dart
  const bool kReleaseMode = bool.fromEnvironment('dart.vm.product');
...
  static late var logger = Logger.detached('Realm')
    ..level = kReleaseMode ? RealmLogLevel.off : RealmLogLevel.info
    ..onRecord.listen((event) => print(event));

```

Reply:
that is working only for flutter no?

Reply:
Let's discuss on the meeting

Reply:
No.. this works for dart as well. But you need to define your own constant.